core-library/src/main/java/org/silverpeas/core/io/media/image/imagemagick/Im4javaImageTool.java

```diff
@@ -149,4 +196,58 @@ private void resize(final IMOperation op,
       }
     }
   }
+
+  /**
+   * Centralizes text watermarking operation
+   * @param op
+   * @param source
+   * @param options
+   * @throws Exception
+   */
+  private void watermarkText(final IMOperation op, final File source,
+      final Map<Class<AbstractImageToolOption>, AbstractImageToolOption> options) throws Exception {
+    WatermarkTextOption watermarkText = getOption(options, WatermarkTextOption.class);
+    if (watermarkText != null) {
+      final String[] imageInfo;
+      DimensionOption dimension = getOption(options, DimensionOption.class);
+      if (dimension != null) {
+        imageInfo = new String[2];
+        imageInfo[0] = dimension.getWidth() != null ? String.valueOf(dimension.getWidth()) :
+            getImageInfo(source, WIDTH_IN_PIXEL)[0];
+        imageInfo[1] = dimension.getHeight() != null ? String.valueOf(dimension.getHeight()) :
+            getImageInfo(source, HEIGHT_IN_PIXEL)[0];
+      } else {
+        imageInfo = getImageInfo(source, WIDTH_IN_PIXEL, HEIGHT_IN_PIXEL);
+      }
+      int width = Integer.valueOf(imageInfo[0]);
+      int height = Integer.valueOf(imageInfo[1]);
+
+      int pointSize = (int) Math.rint((width * 0.02) + Math.log(width));
+      pointSize = Math.min((int) (height * 0.2), pointSize);
+
+      int minX = (int) Math.max(1, (height * 0.015));
+      int minY = (int) Math.max(1, (height * 0.025));
+      int x = (int) (minX + Math.max(1, (pointSize / 2.5)));
+      int y = (int) (minY + Math.max(1, (pointSize / 1.75)));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 1.75 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)